addons/sale/models/sale_order.py

```diff
@@ -914,7 +930,11 @@ def _action_cancel(self):
 
     def _show_cancel_wizard(self):
         for order in self:
-            if order.invoice_ids.filtered(lambda inv: inv.state == 'draft') and not order._context.get('disable_cancel_warning'):
+            if (
+                order.state != 'draft' and not order._context.get('disable_cancel_warning')
+                or order.invoice_ids.filtered(lambda inv: inv.state == 'draft')
+                and not order._context.get('disable_cancel_warning')
+            ):
```

Comment:
```suggestion
               (order.state != 'draft' or order.invoice_ids.filtered(lambda inv: inv.state == 'draft'))
                and not order._context.get('disable_cancel_warning')
            ):
```

Reply:
Ha, my bad, this isn't even necessary, we have the cancel wizard as long as the quotation isn't in draft (and draft quotation don't have invoice) so the check on the invoice is useless anyway